streaming/base/storage/upload.py

```diff
@@ -945,6 +947,137 @@ def check_folder_exists(self):
             raise e
 
 
+class AlipanUploader(CloudUploader):
+    """Upload file from local machine to Alipan.
+
+    Args:
+        out (str | Tuple[str, str]): Output dataset directory to save shard files.
+
+            1. If ``out`` is a local directory, shard files are saved locally.
+            2. If ``out`` is a remote directory, a local temporary directory is created to
+               cache the shard files and then the shard files are uploaded to a remote
+               location. At the end, the temp directory is deleted once shards are uploaded.
+            3. If ``out`` is a tuple of ``(local_dir, remote_dir)``, shard files are saved in
+               the `local_dir` and also uploaded to a remote location.
+        keep_local (bool): If the dataset is uploaded, whether to keep the local dataset
+            shard file or remove it after uploading. Defaults to ``False``.
+        progress_bar (bool): Display TQDM progress bars for uploading output dataset files to
+            a remote location. Default to ``False``.
+        retry (int): Number of times to retry uploading a file. Defaults to ``2``.
+        exist_ok (bool): When exist_ok = False, raise error if the local part of ``out`` already
+            exists and has contents. Defaults to ``False``.
+    """
+
+    def __init__(self,
+                 out: Union[str, Tuple[str, str]],
+                 keep_local: bool = False,
+                 progress_bar: bool = False,
+                 retry: int = 2,
+                 exist_ok: bool = False) -> None:
+        super().__init__(out, keep_local, progress_bar, retry, exist_ok)
+
+        obj = urllib.parse.urlparse(self.remote)
+        if obj.scheme != 'alipan':
+            raise ValueError(
+                f'Expected obj.scheme to be `alipan`, instead, got {obj.scheme} for remote={self.remote}'
+            )
+        if obj.netloc != '':
+            raise ValueError(
+                f'Expected remote is like alipan:///path/to/some, instead, got remote={self.remote}'
```

Comment:
```suggestion
                f'Expected remote to be alipan://path/to/some, instead, got remote={self.remote}'
```

Reply:
The Alipan url doesn't need a `netloc` part. The `path` part is only needed. So I give the url as `alipan:///path/to/some`. Is it ok?